File: src/main/java/org/mystic/codeforces/custom/rockethon/year2014/C1.java
package org.mystic.codeforces.custom.rockethon.year2014;


import java.io.*;
import java.util.ArrayList;
import java.util.Collections;
import java.util.StringTokenizer;

public class C1 implements Runnable {

    PrintWriter out;
    BufferedReader br;
    StringTokenizer st;

    public static void main(String[] args) throws IOException {
        new Thread(new C1()).start();
    }

    public String next() throws IOException {
        while (st == null || !st.hasMoreTokens()) {
            st = new StringTokenizer(br.readLine());
        }
        return st.nextToken();
    }

    public int nextInt() throws IOException {
        return Integer.parseInt(next());
    }

    public long nextLong() throws IOException {
        return Long.parseLong(next());
    }

    public double nextDouble() throws IOException {
        return Double.parseDouble(next());
    }

    public void run() {
        try {
            br = new BufferedReader(new InputStreamReader(System.in));
            out = new PrintWriter(new BufferedOutputStream(System.out));
            solve();
            out.close();
        } catch (IOException e) {
            throw new IllegalStateException(e);
        }
    }

    public void solve() throws IOException {
        final int n = nextInt();
        final int k = nextInt();
        final int[] score = new int[n];
        final int[] powers = new int[n];
        for (int i = 0; i < n; ++i) {
            score[i] = nextInt();
            powers[i] = nextInt();
        }
        int ans = Integer.MAX_VALUE;
        for (int i = 0; i <= Math.pow(2, n); ++i) {
            int tmp = 0;
            int manaoScore = 0;
            final char[] chars = parse(i, n);
            ArrayList<Fighter> s = new ArrayList<>();
            for (int j = 0; j < chars.length; ++j) {
                if (chars[j] == '1') {
                    tmp += powers[j];
                    s.add(new Fighter(j, score[j]));
                    manaoScore++;
                } else {
                    s.add(new Fighter(j, score[j] + 1));
                }

            }
            Collections.sort(s);
            int pos = n + 1;
            for (int j = 0; j < chars.length; ++j) {
                if (manaoScore == s.get(j).score) {
                    if (chars[s.get(j).id] == '1') {
                        pos--;
                    }
                }
                if (manaoScore > s.get(j).score) {
                    pos--;
                }
            }
            if (pos <= k) {
                ans = Math.min(ans, tmp);
            }
        }
        if (ans == Integer.MAX_VALUE) {
            out.println(-1);
        } else {
            out.println(ans);
        }
    }

    private char[] parse(int i, int n) {
        char[] res = new char[n];
        String s = Integer.toBinaryString(i);
        int j;
        for (j = 0; j < n - s.length(); ++j) {
            res[j] = '0';
        }
        for (int k = 0; j < n && k < s.length(); ++j, ++k) {
            res[j] = s.charAt(k);
        }
        return res;
    }

    class Fighter implements Comparable<Fighter> {
        int id;
        int score;

        Fighter(int id, int score) {
            this.id = id;
            this.score = score;
        }

        @Override
        public int compareTo(Fighter o) {
            return Integer.compare(score, o.score);
        }
    }

}

File: src/main/java/org/mystic/ipsc/ipsc2012/C1.scala
package org.mystic.ipsc.ipsc2012

import java.io._
import java.util.{StringTokenizer, TreeMap}

import scala.collection.mutable

object C1 {

  var out: PrintWriter = _
  var br: BufferedReader = _
  var st: StringTokenizer = _

  def main(args: Array[String]): Unit = {
    br = new BufferedReader(new InputStreamReader(System.in))
    out = new PrintWriter(new FileWriter("f2.out"))
    solve
    out.close
  }

  def next: String = {
    while (st == null || !st.hasMoreTokens) {
      st = new StringTokenizer(br.readLine)
    }
    st.nextToken
  }

  def nextInt: Int = Integer.parseInt(next)

  def nextLong: Long = java.lang.Long.parseLong(next)

  def nextDouble: Double = java.lang.Double.parseDouble(next)

  class MultiHashSet[T <% Comparable[T]] {
    val map = new mutable.HashMap[T, Int]()

    def count(x: T): Int = {
      map.getOrElse(x, 0)
    }

    def add(x: T): Unit = map.put(x, count(x) + 1)

    def remove(x: T): Boolean = {
      val prev = count(x)
      if (prev == 0)
        return false
      if (prev == 1) {
        map.remove(x)
      } else {
        map.put(x, prev - 1)
      }
      true
    }
  }

  class MultiTreeSet[T <% Comparable[T]] {
    val map = new TreeMap[T, Int]()

    def count(x: T): Int = {
      map.getOrDefault(x, 0)
    }

    def add(x: T): Unit = map.put(x, count(x) + 1)

    def first(): T = map.firstKey()

    def last(): T = map.lastKey()

    def remove(x: T): Boolean = {
      val prev = count(x)
      if (prev == 0)
        return false
      if (prev == 1) {
        map.remove(x)
      } else {
        map.put(x, prev - 1)
      }
      true
    }
  }

  def pow(a: Int, b: Int): Long = {
    var x: Long = 1L
    for (i <- 1 to b) {
      x *= a.toLong
    }
    x
  }

  def solve: Int = {
    val t = nextInt
    for (i <- 0 until t) {
      val n = nextInt
      val dp = new Array[Double](n)
      dp(0) = 7.000000000d
      for (j <- 1 until n) {
        dp(j) = (dp(j - 1) * Math.floor(dp(j - 1))) / 13.0d
        var sum = 0.0d
        var z = 0
        for (k <- Math.floor(dp(j - 1) + 1).toInt to 13) {
          sum += k
          z += 1
        }
        dp(j) += (sum / z) * ((13.0d - Math.floor(dp(j - 1))) / 13.0d)
      }
      out.println(dp(n - 1))
    }
    0
  }
}
